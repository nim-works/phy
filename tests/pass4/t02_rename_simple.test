discard """
  description: "Rename prohibits copies being performed, unlike Move"
"""
(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Continuations
    (Continuation (Params)
      (Locals (Type 0))
      (Continue 1 (List (Rename (Local 0))))
    )
    (Continuation (Params (Type 0)) (Locals)
      (Continue 2 (List))
    )
    (Continuation (Params)))
  )
)