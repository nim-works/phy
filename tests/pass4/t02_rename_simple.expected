(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Locals (Type 0))
    (Continuations
      (Continuation (Params)
        (Locals (Local 0))
        (Continue 1)
      )
      (Continuation (Params)
        (Locals (Local 0))
        (Continue 2)
      )
      (Continuation (Params))
    )
  )
)
